src/core/server/elasticsearch/retry_call_cluster.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { retryWhen, concatMap } from 'rxjs/operators';
+import { defer, throwError, iif, timer } from 'rxjs';
+import elasticsearch from 'elasticsearch';
+import { CallAPIOptions } from '.';
+
+export function retryCallCluster(
+  callFn: (
```

Comment:
APICaller is incompatible with the internal types in https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/migrations/core/call_cluster.ts once https://github.com/elastic/kibana/pull/46668 is merged, I can update the migrator's types so that we can switch to APICaller.